### PR TITLE
Pull in Camunda Modeler 4.2.0 changes

### LINKED ISSUE #69: Do not allow to model timer/message start event inside embedded sub process
__Describe the Bug__

At the moment the user is able to model a timer/message start event inside an embedded sub process. This should not be allowed.

__Steps to Reproduce__

1. Model an embedded sub process
2. Add a start event to the sub process
3. Morph it to timer/message start event

__Expected Behavior__

The user should only be allowed to model a none start event inside a embedded sub process.

__Environment__

Please complete the following information:

 - OS: Linux
 - Zeeber Modeler Version: 0.6.2

---

Depends on https://github.com/camunda/camunda-modeler/issues/498

### LINKED ISSUE #241: Pull in Camunda Modeler v4.2 changes
__What should we do?__

Pull in [camunda-modeler@4.2.0](https://github.com/camunda/camunda-modeler/issues/1897) changes (once it got released).

```sh
$ npm run sync -- -t 'v4.2.0'
```

__Why should we do it?__

We should be up to date and it also fixes a bug (cf. #69)

------

Child of https://github.com/zeebe-io/zeebe-modeler/issues/232

### PR DESCRIPTION
**Pull request by barmac**
__Which issue does this PR address?__

Closes #241
Closes #69

__Definition of Done__

* [ ] corresponds to [the design principles](https://github.com/bpmn-io/design-principles)
* [ ] corresponds to [the code standards](https://github.com/bpmn-io/bpmn-js/blob/master/.github/CONTRIBUTING.md#creating-a-pull-request)
* [ ] passes Continuous Integration checks
* [ ] available as feature branch on GitHub
  * [ ] contains the cleaned up commit history
  * [ ] commit messages satisfy our [commit message guidelines](https://www.conventionalcommits.org/)
  * [ ] a single commit closes the issue via Closes #issuenr

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
So it seems like we don't use the full commit history (as we did it before, https://github.com/zeebe-io/zeebe-modeler/pull/188), but squashing all changes together?

File: app/lib/file-system.js [hunk 1]
```diff
@@ -36,9 +36,13 @@ const ENCODING_BASE64 = 'base64',
...
  * Read file.
  *
- * @param {String} filePath - Filepath.
+ * @param {string} filePath - Filepath.
  * @param {Object} [options] - Options.
+<<<<<<< HEAD
  * @param {String|false} [options.encoding] - Encoding. Set to false to receive Buffer.
+=======
+ * @param {string} [options.encoding] - Encoding.
+>>>>>>> v4.2.0
```
Comment (pinussilvestrus) [inline app/lib/file-system.js:45]: unresolved merge conflict

Comment (barmac) [reply app/lib/file-system.js:45]: ohhhh gotta fix this

Comment (barmac) [reply app/lib/file-system.js]: Fixed.

**Comment by barmac** [comment]
> So it seems like we don't use the full commit history (as we did it before, #188), but squashing all changes together?

That's what I ended up with due to multiple changes that we do not pull in, e.g. element templates.

On one hand, it would be cool to not have one huge commit, but on the other we can avoid useless commits like https://github.com/zeebe-io/zeebe-modeler/commit/690fc0c482aaa43e1a2ed8b8a958b7b18b7e96da.

**Comment by pinussilvestrus** [comment]
Will we run into trouble when doing the next sync? For example loads of merge conflicts coming in because the commit history of both repositories drove apart too much?

If we won't expect this, I can also live with the huge commit.

**Comment by barmac** [comment]
I don't know. This time, there has already been lots of merge conflicts. I suppose we can try to sync after next Camunda Modeler release so that the changes to merge are relatively small.

**Comment by barmac** [comment]
Anyway, what's the benefit of pulling in the whole Camunda Modeler commit history? We don't include that in the changelog anyway, do we?

File: client/package.json [hunk 2]
```diff
@@ -10,24 +10,24 @@
     "@bpmn-io/align-to-origin": "^0.6.0",
     "@bpmn-io/dmn-migrate": "^0.4.3",
```
Comment (pinussilvestrus) [inline client/package.json:11]: Maybe not part of this PR, because it was there before, but we don't need this dependency in Zeebe Modeler

Comment (barmac) [reply client/package.json:11]: I will remove that then.

**Comment by pinussilvestrus** [comment]
> Anyway, what's the benefit of pulling in the whole Camunda Modeler commit history? We don't include that in the changelog anyway, do we?

I think we have decided on this solution (https://github./camunda/camunda-modeler/issues/1031) to maybe build a sync strategy in the future, where we can exclude certain commits, by using an identifier inside the description or whatever. Therefore a cleaned up history was appreciated.

But I don't think we won't build anything like this in the near future. So I'm fine with the single commit.

File: client/package.json (hunk 2 above)
Comment (pinussilvestrus) [reply client/package.json]: 🚀

**Comment by barmac** [comment]
> We could even consider updating our sync script by not merging the full commit history, but just pull in the changes and let them being unstaged, so we can handle it individually (if that's possible with any git command). Maybe we can talk about it in the next weekly? @barmac

Let's do that!

**Review by pinussilvestrus — APPROVED** [review]

**Pull request merged**